test/e2e/cert_manager_test.go

```diff
@@ -138,3 +143,157 @@ func TestCertManagerProvider(t *testing.T) {
+		var ec ecv1alpha1.EtcdCluster
+		if err := client.Resources().Get(ctx, etcdClusterName, namespace, &ec); err != nil {
+			t.Fatalf("unable to fetch etcd cluster: %s", err)
+		}
+
+		return ctx
+	})
+
+	feature.Assess("Check certificates exist",
+		func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			client := c.Client()
+			// checks if client, server, peer certificates are created in the respective namespace
+			if err := wait.For(
+				func(context.Context) (bool, error) {
+					return validateResourceExists(ctx, client, etcdClusterName, namespace, "certificate")
+				},
+				wait.WithTimeout(3*time.Minute),
+				wait.WithInterval(10*time.Second),
+			); err != nil {
+				t.Fatalf("timed out waiting for certificate: %s", err)
+			}
+
+			return ctx
+		},
+	)
+
+	feature.Assess("Check certificate secrets exist",
+		func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			client := c.Client()
+			// checks if corresponding client, server, peer secrets are created in the respective namespace
+			if err := wait.For(
+				func(context.Context) (bool, error) {
+					return validateResourceExists(ctx, client, etcdClusterName, namespace, "secret")
+				},
+				wait.WithTimeout(3*time.Minute),
+				wait.WithInterval(10*time.Second),
+			); err != nil {
+				t.Fatalf("timed out waiting for certificate: %s", err)
+			}
+			return ctx
+		},
+	)
+
+	_ = testEnv.Test(t, feature.Feature())
+}
+
+func validateResourceExists(ctx context.Context, client klient.Client,
+	etcdClusterName, etcdClusterNamespace, resourceType string) (bool, error) {
+	clientCertName := fmt.Sprintf("%s-client-tls", etcdClusterName)
+	serverCertName := fmt.Sprintf("%s-server-tls", etcdClusterName)
+	peerCertName := fmt.Sprintf("%s-peer-tls", etcdClusterName)
+
+	var obj any
+
+	switch resourceType {
+	case "certificate":
+		var certObj certv1.Certificate
+		obj = &certObj
+	case "secret":
+		var secretObj corev1.Secret
+		obj = &secretObj
+	default:
+		return false, fmt.Errorf("invalid resource type: %v", resourceType)
+	}
+
+	runtimeObj, err := obj.(k8s.Object)
+	if !err {
+		return false, fmt.Errorf("object does not implement runtime.Object: %T", obj)
+	}
+
+	if err := client.Resources().Get(ctx, clientCertName, etcdClusterNamespace, runtimeObj); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to get Client %s: %v", resourceType, err)
+	}
+
+	if err := client.Resources().Get(ctx, serverCertName, etcdClusterNamespace, runtimeObj); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to get Server %s: %v", resourceType, err)
+	}
+
+	if err := client.Resources().Get(ctx, peerCertName, etcdClusterNamespace, runtimeObj); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to get Peer %s: %v", resourceType, err)
+	}
+	return true, nil
```

Comment:
Please also follow the same implementation as below to verify data operations. It might be even better to have a common function to verify data operation, but don't block here for too long, it's OK to enhance this in followup PR. 

https://github.com/etcd-io/etcd-operator/blob/d18a538c53a2e310ce7e9c08603b23f73148d289/test/e2e/e2e_test.go#L331-L352

Reply:
Sure, I'll push the changes within a few hours today for review.

Reply:
> It might be even better to have a common function to verify data operation

Added `verifyDataOperations` which runs the above-mentioned `put/get` check and reused it for validation in `TestClusterCertCreation`. 
In a separate PR, we can refactor this assess block: https://github.com/etcd-io/etcd-operator/blob/main/test/e2e/e2e_test.go#L331-L354 to reuse the `verifyDataOperations`